### PR TITLE
Add fswatch handling to add_on_mod

### PR DESCRIPTION
**Pull request by antevens**
This allows add_on_mod to be used on OSX and platforms without inotify

### CODE REVIEW

**Comment by hrpatel** [comment]
@antevens I assume you've tested this on OSX?

**Comment by antevens** [comment]
I have, but it would be nice if someone else also tested it :)

File: shtdlib.sh [hunk 1]
```diff
@@ -419,18 +419,36 @@ function exit_on_fail {
...
+    elif whichs fswatch ; then
+        file_monitor_command="fswatch  --recursive --format %p
+                                   --event Created
+                                   --event Updated
+                                   --event Removed
+                                   --event Renamed
+                                   --event MovedFrom
+                                   --event MovedTo"
+    else
+        color_echo red "Unable to find inotifywait or fswatch, please install one or the other before trying to use ${0}"
```
Comment (hrpatel) [inline shtdlib.sh]: `${@}` outputs the shell name. 

Suggestion:
```bash
@@ -439,7 +439,7 @@ function add_on_mod {
                                    --event MovedFrom
                                    --event MovedTo"
     else
-        color_echo red "Unable to find inotifywait or fswatch, please install one or the other before trying to use ${0}"
+        color_echo red "Unable to find inotifywait or fswatch, please install one or the other before trying to use '${FUNCNAME[0]} ${*}'"
         return 1
     fi
     local arguments=("${@}")
```

Comment (antevens) [reply shtdlib.sh]: $@ == all arguments including the name of the shell script or if used in the context of a function the function name, if in an interactive shell then the name of the shell
$* == all argumentsincluding the name of the shell script or if used in the context of a function the function name, if in an interactive shell then the name of the shell
$0 == First argument, this is the name of the script  or if used in the context of a function the function name, if in an interactive shell then the name of the shell

Comment (antevens) [reply shtdlib.sh]: Yup

**Pull request merged**